src/cloudant/index.py

```diff
@@ -262,3 +262,91 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes.
+    """
+    def __init__(
+            self,
+            ddoc,
+            search_index_name=None,
+            search_index=None
+    ):
+        super(SearchIndex, self).__init__()
+        self.design_doc = ddoc
+        if search_index_name is not None:
+            self._search_index_name = search_index_name
```

Comment:
I don't understand the purpose of this `if` block or the reason why you default the index name to `None` in the constructor signature.  It does not follow how it is implemented in a `View`.  Can a search index to exist without a name?